pkg/dockerregistry/client.go

```diff
@@ -597,6 +603,88 @@ func (repo *v2repository) getImage(c *connection, image, userTag string) (*Image
 	return repo.getTaggedImage(c, image, userTag)
 }
 
+func (repo *v2repository) getImageConfig(c *connection, dgst string) ([]byte, error) {
+	endpoint := repo.endpoint
+	endpoint.Path = path.Join(endpoint.Path, fmt.Sprintf("/v2/%s/blobs/%s", repo.name, dgst))
+	req, err := http.NewRequest("GET", endpoint.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %v", err)
+	}
+	addAcceptHeader(req)
+
+	if len(repo.token) > 0 {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", repo.token))
+	}
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, convertConnectionError(c.url.String(), fmt.Errorf("error getting image config for %s: %v", repo.name, err))
+	}
+	defer resp.Body.Close()
+
+	switch code := resp.StatusCode; {
+	case code == http.StatusUnauthorized:
+		if len(repo.token) != 0 {
+			// The DockerHub returns JWT tokens that take effect at "now" at second resolution, which means clients can
+			// be rejected when requests are made near the time boundary.
+			if repo.retries > 0 {
+				repo.retries--
+				time.Sleep(time.Second / 2)
+				return repo.getImageConfig(c, dgst)
+			}
+			delete(c.cached, repo.name)
+			// docker will not return a NotFound on any repository URL - for backwards compatibility, return NotFound on the
+			// repo
+			body, _ := ioutil.ReadAll(resp.Body)
+			glog.V(4).Infof("passed valid auth token, but unable to find image config at %q, %d %v: %s", req.URL.String(), resp.StatusCode, resp.Header, body)
+			return nil, errTagNotFound{true, dgst, repo.name}
```

Comment:
This error results in one of the following descriptions based on requested tag:

```go
	if e.wasDefault {
		return fmt.Sprintf("the default tag %q has not been set on repository %q", e.tag, e.repository)
	}
	return fmt.Sprintf("tag %q has not been set on repository %q", e.tag, e.repository)
```

Neither of them apply here. Can we define some other error message or make the `errTagNotFound` more generic?